### PR TITLE
Detailed guide on DIDComm messaging for edge device registration and …

### PR DESCRIPTION
**Pull request by Tekum-Emmanuella**
This article explores DIDComm messaging for secure, communication between edge devices. It details the roles of devices, the Mediator's function in routing messages, and the key phases of communication: Establishing Mediation, Message Exchange Preparation, and Message Pickup and Delivery, emphasizing privacy, security, and flexibility.

### CODE REVIEW

**Comment by Christiantyemele** [comment]
@Tekum-Emmanuella Please can you give a description to this issue

File: docs/edge-device-doc.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#             DIDComm Messaging for Edge Device Registration and Message Exchange
+## Introduction
+With the growth of IoT and connected devices, ensuring secure, private communication between edge devices—like smart sensors, appliances, and mobile devices—has become essential. DIDComm (Decentralized Identifier Communication) provides a protocol for private, out-of-band (OOB) messaging, enabling communication without intermediaries directly accessing data. This makes it a powerful tool for IoT networks where privacy and security are paramount.
```
Comment (Christiantyemele) [inline docs/edge-device-doc.md:3]: **DIDComm provides a protocol for private, out-of-band (OOB) messaging** does didcomm really provides a and out of band messaging way ? It's important to initiate a didcomm communication but didcomm does not provides a way for this to be done

**Review by Christiantyemele — CHANGES_REQUESTED** [review]

**Review by chendiblessing — APPROVED** [review]

File: docs/edge-device-doc.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+* **Phase 1: Establishing Mediation**
+Alice registers with the Mediator to set up secure routing.
+
+* **Mediation Request:** Alice requests mediation services for routing.
+Mediation Grant: The Mediator grants the request, providing Alice with a routing identifier (DID) for secure messaging.
+* **Phase 2: Message Exchange Preparation**
+With mediation in place, Alice prepares to communicate securely:
+
+* **Key Management:** Alice reviews and updates her encryption keys if needed.
+Secure Key Exchange: Alice and the Mediator exchange public keys, establishing the foundation for encrypted communication.
```
Comment (Christiantyemele) [inline docs/edge-device-doc.md:37]: you have to say how alice and the mediator exchange their keys

File: docs/edge-device-doc.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+* **Phase 3: Message Pickup and Delivery**
+Alice and Bob exchange messages with the Mediator’s support:
+
+ * **Message Pickup:** Alice periodically checks with the Mediator for incoming messages.
+* **Message Creation and Delivery:** Alice encrypts a message for Bob, which the Mediator securely forwards without viewing its content.
+* **Acknowledgment:** Bob decrypts and reads Alice’s message, then sends an acknowledgment back to confirm delivery.
```
Comment (Christiantyemele) [inline docs/edge-device-doc.md:43]: it think what you meant is bob prepares a message for a alice, bob cannot receive message here it has no connection with the mediator so far and you have to tell how it is possible for bob to send message to alice through the mediator

**Review by Christiantyemele — CHANGES_REQUESTED** [review]

**Review by ndefokou — APPROVED** [review]
Good :+1:

**Review by Blindspot22 — APPROVED** [review]
LGTM :rocket:

**Review by Christiantyemele — APPROVED** [review]

**Pull request merged**